src/twisted/internet/test/test_gireactor.py

```diff
@@ -103,8 +103,8 @@ def test_gApplicationActivate(self) -> None:
         """
         L{Gio.Application} instances can be registered with a gireactor.
         """
-        reactor = gireactor.GIReactor(useGtk=False)
-        self.addCleanup(self.unbuildReactor, reactor)
+        self.reactorFactory = lambda: gireactor.GIReactor(useGtk=False)
+        reactor = self.buildReactor()
```

Comment:
The refactoring looks very good.

Just a minor suggestion:
```suggestion
        reactor = self.buildReactor(reactorFactory=lambda: gireactor.GIReactor(useGtk=False))
```